packages/ember-htmlbars/tests/helpers/view_test.js

```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
-  view = EmberView.create({
-    customContextView: CustomContextView,
-    template: compile('{{view view.customContextView}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'test');
-});
-
-QUnit.test('Template views add an elementId to child views created using the view helper', function() {
-  owner.register('template:parent', compile('<div>{{view view.childView}}</div>'));
-  owner.register('template:child', compile('I can\'t believe it\'s not butter.'));
-
-  var ChildView = EmberView.extend({
-    templateName: 'child'
-  });
-
-  view = EmberView.create({
-    [OWNER]: owner,
-    childView: ChildView,
-    templateName: 'parent'
-  });
-
-  runAppend(view);
-
-  var childView = get(view, 'childViews.firstObject');
-  equal(view.$().children().first().children().first().attr('id'), get(childView, 'elementId'));
-});
-
-QUnit.test('Child views created using the view helper should have their parent view set properly', function() {
-  var template = '{{#view}}{{#view}}{{view}}{{/view}}{{/view}}';
-
+QUnit.test('Should not apply classes when bound property specified is false', function() {
   view = EmberView.create({
-    template: compile(template)
+    controller: {
+      someProp: false
+    },
+    template: compile('{{#view id="foo" class=someProp}} Foo{{/view}}')
   });
 
   runAppend(view);
 
-  var childView = firstGrandchild(view);
-  equal(childView, get(firstChild(childView), 'parentView'), 'parent view is correct');
+  ok(!jQuery('#foo').hasClass('some-prop'), 'does not add class when value is falsey');
 });
 
-QUnit.test('Child views created using the view helper should have their IDs registered for events', function() {
-  var template = '{{view}}{{view id="templateViewTest"}}';
-
+QUnit.test('Should apply classes of the dasherized property name when bound property specified is true', function() {
   view = EmberView.create({
-    template: compile(template)
+    controller: {
+      someProp: true
+    },
+    template: compile('{{#view id="foo" class=someProp}} Foo{{/view}}')
   });
 
   runAppend(view);
 
-  var childView = firstChild(view);
-  var id = childView.$()[0].id;
-  equal(EmberView.views[id], childView, 'childView without passed ID is registered with View.views so that it can properly receive events from EventDispatcher');
-
-  childView = nthChild(view, 1);
-  id = childView.$()[0].id;
-  equal(id, 'templateViewTest', 'precond -- id of childView should be set correctly');
-  equal(EmberView.views[id], childView, 'childView with passed ID is registered with View.views so that it can properly receive events from EventDispatcher');
+  ok(jQuery('#foo').hasClass('some-prop'), 'adds dasherized class when value is true');
 });
 
-QUnit.test('Child views created using the view helper and that have a viewName should be registered as properties on their parentView', function() {
-  var template = '{{#view}}{{view viewName="ohai"}}{{/view}}';
-
+QUnit.test('should escape HTML in normal mustaches', function() {
```

Comment:
Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR660